files/plir-openhim/core-config.json

```diff
@@ -0,0 +1,154 @@
+      "externalHostname": "localhost",
+      "httpPort": 5001,
+      "httpsPort": 5000,
+      "timeout": 60000
+    },
+    "bodyCull":{
+      "enabled":true,
+      "pollPeriodMins": 60
+    },
+    "api": {
+      "enabled": true,
+      "protocol": "https",
+      "port": 8080,
+      "authWindowSeconds": 10,
+      "maxBodiesSizeMB": 15,
+      "maxPayloadSizeMB": 50,
+      "truncateSize": 15000,
+      "truncateAppend": "\n[truncated ...]",
+      "authenticationTypes": ["token"]
+    },
+    "rerun": {
+      "httpPort": 7786,
+      "host": "localhost",
+      "processor": {
+        "enabled": true,
+        "pollPeriodMillis": 2000
+      }
+    },
+    "tcpAdapter": {
+      "httpReceiver": {
+        "httpPort": 7787,
+        "host": "localhost"
+      }
+    },
+    "polling": {
+      "enabled": true,
+      "pollingPort": 7788,
+      "host": "localhost"
+    },
+    "authentication": {
+      "enableMutualTLSAuthentication": true,
+      "enableBasicAuthentication": true,
+      "enableCustomTokenAuthentication": false,
+      "enableJWTAuthentication": false,
+      "jwt": {
+        "secretOrPublicKey": "",
+        "algorithms": "",
+        "audience": "",
+        "issuer": ""
+      }
+    },
+    "email": {
+      "fromAddress": "address@example.com",
+      "nodemailer": {
+        "service": "Gmail",
+        "auth": {
+          "user": "user@gmail.com",
+          "pass": "password"
+        }
+      }
+    },
+    "smsGateway": {
+      "provider": "clickatell",
+      "config": {
+        "user": "user",
+        "pass": "pass",
+        "apiID": "apiID"
+      }
+    },
+    "alerts": {
+      "enableAlerts": true,
+      "pollPeriodMinutes": 1,
+      "himInstance": "Local OpenHIM",
+      "consoleURL": "http://localhost:9000"
+    },
+    "reports": {
+      "enableReports": true,
+      "dailyReportAt": "0 7 * * *",
+      "weeklyReportAt": "0 7 * * 1",
+      "utcOffset": null
+    },
+    "events": {
+      "enableTSNormalization": true
+    },
+    "newUserExpiry": {
+      "duration": 7,
+      "durationType": "days"
+    },
+    "userPasswordResetExpiry": {
+      "duration": 1,
+      "durationType": "hours"
+    },
+    "auditing": {
+      "servers": {
+        "udp": {
+          "enabled": false,
+          "port": 5050
+        },
+        "tls": {
+          "enabled": false,
+          "port": 5051
+        },
+        "tcp": {
+          "enabled": false,
+          "port": 5052
+        }
+      },
+      "auditEvents": {
+        "interface": "internal",
+        "host": "localhost",
+        "port": 5051,
+        "auditSourceID": "OpenHIM"
+      }
+    },
+    "caching": {
+      "enabled": true,
+      "refreshMillis": 1000
+    },
+    "tlsClientLookup": {
+      "type": "strict"
+    },
+    "agenda": {
+      "startupDelay": 500
+    },
+    "certificateManagement": {
+      "watchFSForCert": false,
+      "certPath": "/etc/letsencrypt/live/namanga.openmrs.org/fullchain.pem",
```

Comment:
@PermissionError , i set the path here

Reply:
I think we have to mount the certificate folder into the docker container first, since openhim is in a docker container it can't access files on the host machine without mounts.

Reply:
oh ya makes sense

Reply:
I think the first question here is _why_ do you need it. 

We try to leave nginx to do the offloading of TLS/SSL, and leave this complexity out of containers. 

If you can't, for any reason, it gets annoying. 

You can check the ldap container, but then every time that letsencrypt change certs, I need to force restart LDAP. 

Can we not put SSL in the docker containers at all and leave that work for nginx?

Reply:
@cintiadr   ,
not so sure about that , but for some reason , the transaction in openhim was failing because of the core using a self assigned certificate.

```
Error: self signed certificate
 at TLSSocket.onConnectSecure (_tls_wrap.js:1321:34)
 at TLSSocket.emit (events.js:223:5)
 at TLSSocket._finishInit (_tls_wrap.js:794:8)
 at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:608:12)
```

Reply:
By default the Openhim core requires SSL Auth ,
i have instead disabled it